packages/design-system/src/components/NumericInputField/NumericInput.js

```diff
@@ -9,12 +9,15 @@ const cx = classNames.bind(styles);
 const baseClass = 'numeric-input';
 
 class NumericInput extends React.PureComponent {
+  hasMin = () => this.props.min !== undefined;
+  hasMax = () => this.props.max !== undefined;
+  
   componentDidMount() {
-    if (this.props.max && parseInt(this.props.value, 10) > this.props.max) {
+    if (this.hasMax() && parseInt(this.props.value, 10) > this.props.max) {
```

Comment:
I found it a little bit weird that this `min` `max` values validation is happening on `componentDidMount` only. There is actually possibility that those values will change meanwhile component lifecycle. So, in that case, it could be done additionally in `componentDidUpdate` method or by using the derived state to keep those validated values in sync with raw props.